repl/src/test/scala/ammonite/repl/ScriptTests.scala

```diff
@@ -299,6 +303,55 @@ object ScriptTests extends TestSuite{
             assert(n == expected)
           }
         }
+
+        'processModuleCaching{
+          def check(script: RelPath){
+            val storage = new Storage.Folder(tempDir)
+            val interp1 = createTestInterp(
+              storage,
+              ammonite.repl.Main.defaultPredefString
+            )
+            interp1.replApi.load.module(resourcesPath/script)
+            assert(interp1.compiler != null)
+            val interp2 = createTestInterp(
+              storage,
+              ammonite.repl.Main.defaultPredefString
+            )
+            assert(interp2.compiler == null)
+            interp2.replApi.load.module(resourcesPath/script)
+            assert(interp2.compiler == null)
+          }
+
+          'testOne - check('scriptLevelCaching/"scriptTwo.scala")
+          'testTwo - check('scriptLevelCaching/"scriptOne.scala")
+          'testThree - check('scriptLevelCaching/"QuickSort.scala")
+          'testLoadModule - check('scriptLevelCaching/"testLoadModule.scala")
+          'testFileImport - check('scriptLevelCaching/"testFileImport.scala")
```

Comment:
We should have another test here to use `import $ivy` to verify that that keeps working inside cached scripts